### ISSUE TITLE
MsalInterceptor renews token but Authorization header is missing from intercepted call

### ISSUE THREAD

**Issue by jukvaa** [AzureAD/microsoft-authentication-library-for-js#1475]
## Library
- [x] `msal@1.2.2`
- [x] `@azure/msal-angular@1.0.0-beta.4`
- [x] `@angular@9.0.1`

## Framework
Angular 9

## Description
We have angular front and are using @azure/msal-angular@1.0.0-beta.4. After an hour after login when token expires, next api calls fails, because it has null in Authorization header:

`Authorization: Bearer null`

Renewal itself seems to work because api calls after this failed one work ok. Shouldn't MsalInterceptor renew token and apply it to the intercepted request?

This is msal log from the renewal:

```
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:43 GMT:1.2.2-Info renewidToken is called
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:43 GMT:1.2.2-Info Add msal frame to document:msalIdTokenFrame
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:43 GMT:1.2.2-Info LoadFrame: msalIdTokenFrame
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:44 GMT:1.2.2-Info Add msal frame to document:msalIdTokenFrame
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:44 GMT:1.2.2-Info Returned from redirect url
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:44 GMT:1.2.2-Info Processing the callback from redirect response
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:44 GMT:1.2.2-Info State status:true; Request type:LOGIN
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:44 GMT:1.2.2-Info State is right
main-es2015.f4c8c8ef197a06fb7d31.js:1 MSAL:  Tue, 07 Apr 2020 08:11:44 GMT:1.2.2-Info Fragment has id token
```

## Security
No

## Regression
Unknown

## Configuration

```
{
      auth: {
        clientId: environment.clientId,
        authority: 'https://login.microsoftonline.com/' + environment.tenantId,
      },
      cache: {
        cacheLocation: 'sessionStorage',
        storeAuthStateInCookie: isIE
      },
    }, 

    {
        unprotectedResources: [
          '/xxx/yyy',
        ]
    }
```

## Reproduction steps
1. Login to angular application using AAD and msal-angular
2. Wait an hour for the token to expire
3. Initiate api call from frontend.

## Expected behavior
token renewal should happen in background and api call should not fail.

**Comment by oleksii-kostin** [comment]
I`m experiencing this issue as well. 
msal.interceptor.ts 
```
        return from(
            this.auth.acquireTokenSilent({ scopes })
                .then((response: AuthResponse) => {
                    accessToken = response.accessToken;
                    const authHeader = `Bearer ${response.accessToken}`;
                    return req.clone({
                        setHeaders: {
                            Authorization: authHeader,
                        }
                    });
                })
        )
```
response.accessToken is null, as now response is object

**Comment by jasonnutter** [comment]
@jukvaa @oleksii-kostin This is a known issue (ID token being retrieved instead of an access token), and we'll have a fix for this soon.

**Comment by thesalex** [comment]
I have the same issue, updated to msal 1.3.0 and msal-angular beta5 buy the issue persist.
The fix Will Be released in the future?
Thanks for the support.

**Comment by jasonnutter** [comment]
@thesalex Yes, we'll have a fix in the future. For now, a workaround is to retry the http request when this error occurs, which should succeed.

**Comment by mikeyuan2008** [comment]
We got simular 401 error in our QA/PROD ENV when call API, and API got: Authorization: Bearer null

But we have different MSAL output from the Console Log, we tried but not able to reproduce the issue described here. . 

@jasonnutter is it possible for you to help explain what's the scenario and how can we reproduce this error based on the Console Output here? 

We used msal-angular 1.0.0-beta.5 + msal 1.3.0-beta.0. (we not update to small 1.3.0 + meal-angular 1.0.0 yet)

MSAL Logging:  Wed, 29 Apr 2020 21:24:05 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Add msal frame to document:msalRenewFrame ...
MSAL Logging:  Wed, 29 Apr 2020 21:24:05 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info LoadFrame: msalRenewFrame...
MSAL Logging:  Wed, 29 Apr 2020 21:24:06 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Add msal frame to document:msalRenewFrame...
MSAL Logging:  Wed, 29 Apr 2020 21:24:07 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Processing the callback from redirect response
MSAL Logging:  Wed, 29 Apr 2020 21:24:07 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info State status:true; Request type:RENEW_TOKEN
MSAL Logging:  Wed, 29 Apr 2020 21:24:07 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info State is right
MSAL Logging:  Wed, 29 Apr 2020 21:24:07 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Fragment has access token
MSAL Logging:  Wed, 29 Apr 2020 21:24:07 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info The user object received in the response is the same as the one passed in the acquireToken request
MSAL Logging:  Wed, 29 Apr 2020 21:24:49 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:...
MSAL Logging:  Wed, 29 Apr 2020 21:24:49 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:
MSAL Logging:  Wed, 29 Apr 2020 21:24:49 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:
MSAL Logging:  Wed, 29 Apr 2020 21:24:49 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:...
MSAL Logging:  Wed, 29 Apr 2020 21:24:49 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:...
MSAL Logging:  Wed, 29 Apr 2020 21:24:49 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:...
MSAL Logging:  Wed, 29 Apr 2020 21:24:49 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:...
MSAL Logging:  Wed, 29 Apr 2020 21:25:50 GMT:1de1ebdd-ab54-42a2-bb9e-83d87b9a69da-1.3.0-Info Token is already in cache for scope:...
api/program?_=1588195550536:1 Failed to load resource: the server responded with a status of 401 (Unauthorized)

**Comment by jasonnutter** [comment]
@mikeyuan2008 This was fixed in `@azure/msal-angular@1.0.0`, please upgrade and let me know if it is still an issue. I realize this isn't reflected in the changelog, I'll make sure that gets updated.

**Comment by srivishalsangepu** [comment]
> @thesalex Yes, we'll have a fix in the future. For now, a workaround is to retry the http request when this error occurs, which should succeed.

Hi @jasonnutter , I'm facing the similar issue, token renewal is failing, but If I refresh the page it is taking the token and logging me in. Is there any way to resolve the issue, other than page refresh/reload. Like you were talking about some http retry, which http call you are talking about. Can you please help me in resolving the issue.
Thanks

**Comment by mikeyuan2008** [comment]
> @mikeyuan2008 This was fixed in `@azure/msal-angular@1.0.0` (you should also update to `msal@1.3.0`), please upgrade and let me know if it is still an issue. I realize this isn't reflected in the changelog, I'll make sure that gets updated.
> 
> > @jasonnutter is it possible for you to help explain what's the scenario and how can we reproduce this error based on the Console Output here?
> 
> This can occur when the clientId of your application is set as the scope for a resource, which results in an ID token being fetched, instead of an access token. This could be done manually in the `protectedResourceMap`, or as [the fallback logic](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/src/msal.service.ts#L204-L214) inside `getScopesForEndpoint`.
> 
> Previously, the interceptor naively assumed that `acquireTokenSilent` would always return an access token, but in this scenario, `response.accessToken` would be `null` and the response would instead include only an ID token. While we don't recommend protecting resources with ID tokens, it is technically possible to do, which is why the fix was made even though its not the recommended best practice for protecting resources.

@jasonnutter We applied the MSAL1.3.0+MSALAngular1.0.0 and the issue still there. We did further testing and analyze, here is the behaviour we observed:
- Only happens for user who didn't use the system for long time (2 weeks). We used B2C and have Refresh token set as 14days)
- Error happen in such scenarios: user login successful, app send few Get API requests to load some data, then user try to post a request to create an object, at that time, got 401 error
- We added more log in API, and find the Authentication header is null when 401 error happens. 
- From the App Insight Logs and Console verbose logs, we found MSAL found the token from cache and use it for Http Interceptor. But somehow the access token from cache entry is null. is it possible in some scenario the cache entry is there, but the access token inside cache entry is null?

Interesting thing is:
. Only for user did not use system for long time. Never happen for developer/QA who keep use the system.
. Only happens when user try to create with Http Post, the Http Get request which happens few seconds before works fine. 
. Not able to reproduce it, the error is gone if try again, that is because in MSAL library, when 401 error return, MSAL clean up the cache.

**Comment by jasonnutter** [comment]
@jukvaa We're aware of this happening after long periods inactivity, should be addressed by #1681
